lib/portfolio/widgets/asset_list_tile.dart

```diff
@@ -13,11 +15,25 @@ class AssetListTile extends HookConsumerWidget {
   }) : super(key: key);
 
   final AssetModel asset;
+  double updateDayVariation() {
+    asset.variation = (dtd(asset.prices[0]) / dtd(asset.prices[1]) - 1) * 100;
+    return asset.variation;
+  }
 
   @override
   Widget build(BuildContext context, WidgetRef ref) {
     var visible = ref.watch(visibleProvider.state);
     return ListTile(
+      onTap: () {
+        asset.variation = updateDayVariation();
+        Navigator.pushNamed(
+          context,
+          '/details',
+          arguments: Arguments(
+            asset: asset,
+          ),
```

Comment:
MaterialApp(
  // Provide a function to handle named routes.
  // Use this function to identify the named
  // route being pushed, and create the correct
  // Screen.
  onGenerateRoute: (settings) {
    // If you push the PassArguments route
    if (settings.name == PassArgumentsScreen.routeName) {
      // Cast the arguments to the correct
      // type: ScreenArguments.
      final args = settings.arguments as ScreenArguments;

      // Then, extract the required data from
      // the arguments and pass the data to the
      // correct screen.
      return MaterialPageRoute(
        builder: (context) {
          return PassArgumentsScreen(
            title: args.title,
            message: args.message,
          );
        },
      );
    }
    // The code only supports
    // PassArgumentsScreen.routeName right now.
    // Other values need to be implemented if we
    // add them. The assertion here will help remind
    // us of that higher up in the call stack, since
    // this assertion would otherwise fire somewhere
    // in the framework.
    assert(false, 'Need to implement ${settings.name}');
    return null;
  },
) 
o ideal seria fazer no main app